### PR TITLE
Use config-independent-endianness conversion functions

### PR DESCRIPTION
**Pull request by rnestler**
Check the following:

 - [na] Breaking changes marked in commit message
 - [x] Changelog updated
 - [x] Code style cleaned (ran `make style-fix`)
 - [x] Tested on actual hardware

### CODE REVIEW

File: sps30-i2c/sps30.c [hunk 1]
```diff
@@ -118,70 +118,55 @@ int16_t sps30_read_data_ready(uint16_t *data_ready) {
...
-    SENSIRION_WORDS_TO_BYTES(data.u16_value,
-                             SENSIRION_NUM_WORDS(data.u16_value));
+    error = sensirion_i2c_read_words_as_bytes(SPS30_I2C_ADDRESS, data,
+                                              SENSIRION_NUM_WORDS(data));
+    if (error != NO_ERROR) {
+        return error;
+    }
 
-    *interval_seconds = be32_to_cpu(data.u32_value);
+    *interval_seconds = sensirion_bytes_to_float(data);
```
Comment (abrauchli) [inline sps30-i2c/sps30.c:169]: this one is not a float, use `sensirion_bytes_to_uint32_t`

Comment (rnestler) [reply sps30-i2c/sps30.c:169]: :facepalm:

Comment (rnestler) [reply sps30-i2c/sps30.c]: Fixed.

Comment (rnestler) [reply sps30-i2c/sps30.c]: At least this one was caught by CI as well :slightly_smiling_face:

**Comment by rnestler** [comment]
I noticed that I missed one use of the legacy endianness conversion macros (While compiling on a RaspberryPi 1 on Rasbpian Jessy since the old gcc version they have assumes C89 per default and the macro is violating that).
https://github.com/Sensirion/embedded-sps/blob/e471f1322b90827736b417d844abc029dfc64fcb/sps30-i2c/sps30.c#L86

**Pull request merged**